wire-ios-request-strategy/Sources/Message Sending/MessageDependencyResolver.swift

```diff
@@ -32,41 +35,55 @@ public class MessageDependencyResolver: MessageDependencyResolverInterface {
 
     let context: NSManagedObjectContext
 
-    public func waitForDependenciesToResolve(for message: any SendableMessage) async {
+    public func waitForDependenciesToResolve(for message: any SendableMessage) async -> Swift.Result<Void, MessageDependencyResolverError> {
+
+        @Sendable func dependenciesAreResolved() -> Swift.Result<Bool, MessageDependencyResolverError> {
+            if (self.context.performAndWait {
+                message.conversation?.securityLevel == .secureWithIgnored
+            }) {
+                return .failure(MessageDependencyResolverError.securityLevelDegraded)
+            }
 
-        @Sendable func dependenciesAreResolved() -> Bool {
             let hasDependencies = self.context.performAndWait {
                 message.dependentObjectNeedingUpdateBeforeProcessing != nil
             }
 
             if !hasDependencies {
                 WireLogger.messaging.debug("Message dependency resolved")
-                return true
+                return .success(true)
             } else {
                 WireLogger.messaging.debug("Message has dependency, waiting")
-                return false
+                return .success(false)
             }
         }
 
-        if dependenciesAreResolved() {
-            return
+        let result = dependenciesAreResolved()
+        let continueWaiting = result.isOne(of: .success(false))
+
+        if !continueWaiting {
+            return result.map { _ in Void() }
         }
 
-        await withCheckedContinuation { continuation in
+        return await withCheckedContinuation { continuation in
             Task {
-                if dependenciesAreResolved() {
-                    continuation.resume()
+                let result = dependenciesAreResolved()
+                let continueWaiting = result.isOne(of: .success(false))
+
+                if !continueWaiting {
+                    continuation.resume(returning: result.map({ _ in Void() }))
```

Comment:
Why does this need to be repeated, if we already know that dependencies are not yet resolved?

Reply:
turns out it uncessary this all got simplified in https://github.com/wireapp/wire-ios/pull/659/commits/16948f7d59e17adfaca9183e8e3984a6c449d056